src/Path/Include.hs

```diff
@@ -409,6 +408,61 @@ infixr 7 -<.>
   -> m (Path b File)   -- ^ New file name with the desired extension
 (-<.>) = flip setFileExtension
 
+-- | Tries to remove specific extension from given file path. Throws if the
+-- resulting filename does not parse.
+--
+-- >>> stripFileExtension "txt" $(mkRelFile "foo")
+-- Nothing
+-- >>> stripFileExtension "baz" $(mkRelFile "foo.bar.baz")
+-- Just "foo.bar"
+-- >>> stripFileExtension ".baz" $(mkRelFile "foo.bar.baz")
+-- Just "foo.bar"
+-- >>> stripFileExtension "bar.baz" $(mkRelFile "foo.bar.baz")
+-- Just "foo"
+-- >>> stripFileExtension ".bar.baz" $(mkRelFile "foo.bar.baz")
+-- Just "foo"
+-- >>> stripFileExtension "foo.bar.baz" $(mkRelFile ".foo.bar.baz")
+-- *** Exception: InvalidRelFile ""
+--
+-- @since 0.6.2
+stripFileExtension
+  :: MonadThrow m
+  => String                  -- ^ Extension to strip
+  -> Path b File             -- ^ Old file name
+  -> m (Maybe (Path b File)) -- ^ New file name without extension
+stripFileExtension ext =
+#if MIN_VERSION_filepath(1,4,1)
+  liftFilePathFunc (FilePath.stripExtension ext)
+#else
+  liftFilePathFunc stripExt
```

Comment:
There is risk of these two implementations getting out of sync if the filepath implementation changes. To avoid that we can perhaps just remove the CPP conditional and keep the full implementation here.

Reply:
True, but then if the `filepath` package moves on then `path`'s behaviour would diverge, which doesn't seem right to me.

Reply:
Whenever we use the `filepath` routines we are saying that the behavior of `path` depends on the version of `filepath` you are choosing. So the behavior of path may change even without a change of version, this is a bit disconcerting. This is a general problem though since we are using `filepath` routines at other places as well.